pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -907,6 +927,18 @@ type KubeAPIServerConfig struct {
 	// This field is only available for Kubernetes v1.30 or later.
 	// +optional
 	StructuredAuthorization *StructuredAuthorization `json:"structuredAuthorization,omitempty" protobuf:"bytes,18,opt,name=structuredAuthorization"`
+	// Autoscaling contains auto-scaling configuration options for the kube-apiserver.
+	// +optional
+	Autoscaling *ControlPlaneAutoscaling `json:"autoscaling,omitempty" protobuf:"bytes,19,opt,name=autoscaling"`
+}
+
+// ControlPlaneAutoscaling contains auto-scaling configuration options for control-plane components.
+type ControlPlaneAutoscaling struct {
+	// MinAllowed configures the minimum allowed resource requests.
+	// Configuration of minAllowed resources is an advanced feature that can help clusters to overcome scale-up delays.
+	// Default values are not applied to this field.
+	// +optional
+	MinAllowed corev1.ResourceList `json:"minAllowed,omitempty" protobuf:"bytes,1,rep,name=minAllowed,casttype=k8s.io/api/core/v1.ResourceList,castkey=k8s.io/api/core/v1.ResourceName"`
```

Comment:
Should this really be optional? It is the only field in this struct, so perhaps it's mandatory?

Reply:
Do you see any benefit of making is mandatory? My impression is that it doesn't matter too much in this case and it's fine to leave it optional 🤔

Reply:
The only benefit I see is that making it mandatory prevents people from configuring this with an empty `ControlPlaneAutoscaling` struct (right?), but yes, I agree that the impact is limited.